channels/rdpecam/client/camera_device_main.c

```diff
@@ -24,11 +24,14 @@
 #define TAG CHANNELS_TAG("rdpecam-device.client")
 
 /* supported formats in preference order:
- * passthrough, I420 (used as input for H264 encoder), other YUV based, RGB based
+ * H264, MJPG, I420 (used as input for H264 encoder), other YUV based, RGB based
+ *
+ * comment out any line to disable specific format for testing
  */
 static const CAM_MEDIA_FORMAT_INFO supportedFormats[] = {
```

Comment:
Just wondering the rational of that array, as with any format we end up with doing h264 at the end...

Reply:
The `outputFormat` column in this array serves two purposes:

1. If `inputFormat` equals `outputFormat`, we do the pass through (no encoding):
https://github.com/FreeRDP/FreeRDP/blob/78044ab1b44c49774749fcc7d29b0fb9ae5a6b4b/channels/rdpecam/client/camera_device_main.c#L142
2. The MS-RDPECAM spec doesn't limit the output format to H264 only. For practical purposes (bandwidth), we keep it like that, but in theory we can have:
```
	{ CAM_MEDIA_FORMAT_MJPG, CAM_MEDIA_FORMAT_MJPG },
```
in which case we will pass through MJPG. Also, MS may add other high compression codecs in future (H265, etc), so we will be ready for that :)